modules/solid_mechanics/src/timeintegrators/ExplicitMixedOrder.C

```diff
@@ -0,0 +1,341 @@
+#include "MooseTypes.h"
+#include "MooseVariableFieldBase.h"
+#include "NonlinearSystem.h"
+#include "FEProblem.h"
+#include "TimeStepper.h"
+#include "TransientBase.h"
+
+// libMesh includes
+#include "TransientBase.h"
+#include "libmesh/id_types.h"
+#include "libmesh/nonlinear_solver.h"
+#include "libmesh/sparse_matrix.h"
+#include "DirichletBCBase.h"
+#include "libmesh/vector_value.h"
+#include <algorithm>
+#include <iterator>
+#include <utility>
+
+registerMooseObject("SolidMechanicsApp", ExplicitMixedOrder);
+registerMooseObjectRenamed("SolidMechanicsApp",
+                           DirectCentralDifference,
+                           "10/14/2025 00:00",
+                           ExplicitMixedOrder);
+
+InputParameters
+ExplicitMixedOrder::validParams()
+{
+  InputParameters params = ExplicitTimeIntegrator::validParams();
+
+  params.addClassDescription(
+      "Implementation of explicit time integration without invoking any of the nonlinear solver.");
+
+  params.addParam<bool>("use_constant_mass",
+                        false,
+                        "If set to true, will only compute the mass matrix in the first time step, "
+                        "and keep using it throughout the simulation.");
+  params.addParam<TagName>("mass_matrix_tag", "mass", "The tag for the mass matrix");
+
+  params.addParam<std::vector<VariableName>>(
+      "second_order_vars",
+      {},
+      "A subset of variables that require second-order integration (velocity and acceleration) to "
+      "be applied by this time integrator.");
+
+  params.addParam<std::vector<VariableName>>(
+      "first_order_vars",
+      {},
+      "A subset of variables that require first-order integration (velocity only) to be applied by "
+      "this time integrator.");
+
+  // Prevent users from using variables option by accident.
+  params.suppressParameter<std::vector<VariableName>>("variables");
+
+  MooseEnum solve_type("consistent lumped lump_preconditioned", "lumped");
+  params.setParameters("solve_type", solve_type);
+  params.ignoreParameter<MooseEnum>("solve_type");
+  return params;
+}
+
+ExplicitMixedOrder::ExplicitMixedOrder(const InputParameters & parameters)
+  : ExplicitTimeIntegrator(parameters),
+    _constant_mass(getParam<bool>("use_constant_mass")),
+    _mass_matrix(getParam<TagName>("mass_matrix_tag")),
+    _solution_older(_sys.solutionState(2)),
+    _vars_first(declareRestartableData<std::unordered_set<unsigned int>>("first_order_vars")),
+    _local_first_order_indicies(
+        declareRestartableData<std::vector<dof_id_type>>("first_local_indices")),
+    _vars_second(declareRestartableData<std::unordered_set<unsigned int>>("second_order_vars")),
+    _local_second_order_indicies(
+        declareRestartableData<std::vector<dof_id_type>>("second_local_indices"))
+{
+  _fe_problem.setUDotRequested(true);
+  _fe_problem.setUDotOldRequested(true);
+  _fe_problem.setUDotDotRequested(true);
+  _fe_problem.setUDotDotOldRequested(true);
```

Comment:
do you always need those? If all first order for example?

Reply:
Yes for now we are always requesting all of them because in `performExplicitSolve` we always grab it without branching if a problem is only first order.